uwblueprint/humane-society
# Added user_behaviours table and updated pet_behaviours table

## Notion ticket link
[Ticket Name](https://www.notion.so/uwblueprintexecs/Extend-Behaviours-Entity-II-10d10f3fb1dc8093aa66e131db07dfd2)



## Implementation description
* Added `user_behaviours` model and migration file
* Added columns to `pet_behaviours` table



## Steps to test
1. Run `node migrate up`, should have 13 rows
2. Run `node migrate down` twice should have 12 rows

## Tests
For `pet_behaviours`:
1. Throwing errors when `pet_id` or  `behaviour_id` doesn't exist.
2. Throwing errors when unique constraint is broken  (combination of `pet_id` and  `behaviour_id` should be unique)
3. Throwing errors when `skill_level` is not in the range 1 - 4.
4. Success adding a row when all constraints are met.

For `user_behaviours`:
1. Throwing errors when `user_id` or  `behaviour_id` doesn't exist.
2. Throwing errors when unique constraint is broken (combination of `user_id` and  `behaviour_id` should be unique)
3. Throwing errors when `max_level` is not in the range 1 - 4.
4. Success adding a row when all constraints are met.

For `behaviour_level_details`:
1. Throwing errors when `behaviour_id` doesn't exist.
2. Throwing errors when unique constraint is broken (combination of `level` and  `behaviour_id` should be unique)
3. Throwing errors when `level` is not in the range 1 - 4.
4. Success adding a row when all constraints are met.


## What should reviewers focus on?
* Constraints for unique pet behaviour


## Checklist
